priam/src/main/java/com/netflix/priam/backup/IBackupFileSystem.java

```diff
@@ -114,6 +114,16 @@ void uploadFile(
             final boolean deleteAfterSuccessfulUpload)
             throws FileNotFoundException, RejectedExecutionException, BackupRestoreException;
 
+    /**
+     * Get the bucket where this object should be stored. For local file system this should be empty
+     * or null.
+     *
+     * @return the location of the bucket.
+     */
+    default String getBucket() {
```

Comment:
Is this leaking the implementation details (S3) into the interface? What if it's just `getShard` or something?